packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -0,0 +1,110 @@
+import type { TSESLint } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+
+import { createRule, getParserServices } from '../util';
+
+export default createRule({
+  name: 'prefer-find',
+  meta: {
+    docs: {
+      description:
+        'Enforce the use of Array.prototype.find() over Array.prototype.filter() followed by [0] when looking for a single result',
+      requiresTypeChecking: true,
+      recommended: 'stylistic',
+    },
+    fixable: 'code',
+    messages: {
+      preferFind: 'Use .filter(...)[0] instead of .find(...)',
+      preferFindFix: 'Use .filter(...)[0] instead of .find(...)',
```

Comment:
[Style] Just nitpicking, I assumed reading this at first to be an autofix/`--fix` rather than a suggestion. Maybe:

```suggestion
      preferFindSuggestion: 'Use .filter(...)[0] instead of .find(...)',
```

If you're going off existing precedent then totally ignore this. I do wonder if we should make a lint rule in [`eslint-plugin-eslint-plugin`](https://github.com/eslint-community/eslint-plugin-eslint-plugin). WDYT?

Reply:
Changed the messageIds.

I do have a question though of whether the messages themselves are ok, or need to be more thought out. I guess at a minimum I should probably make it use `.at()`.... or, make the wording _less_ specific, i.e. `Use Array.prototype.find() rather than accessing the first element of Array.prototype.filter()`. 

In any case, please let me know what's expected here! Thanks :)